internal/handlers/common/update_array_operators.go

```diff
@@ -414,3 +415,66 @@ func processPullAllArrayUpdateExpression(doc, update *types.Document) (bool, err
 
 	return changed, nil
 }
+
+// processPullArrayUpdateExpression changes document according to $pull array update operator.
+// If the document was changed it returns true.
+func processPullArrayUpdateExpression(doc *types.Document, update *types.Document) (bool, error) {
+	var changed bool
+
+	iter := update.Iterator()
```

Comment:
I think we still need `defer iter.Close()` even though in this particular case it doesn't hurt.

Reply:
Fixed it here and in a couple of other places.